pkg/reconciler/clusteringress/clusteringress.go

```diff
@@ -61,21 +69,63 @@ var (
 
 // Reconciler implements controller.Reconciler for ClusterIngress resources.
 type Reconciler struct {
-	*reconciler.Base
-
-	// listers index properties about resources
+	*ing.BaseIngressReconciler
 	clusterIngressLister listers.ClusterIngressLister
-	virtualServiceLister istiolisters.VirtualServiceLister
-	gatewayLister        istiolisters.GatewayLister
-	secretLister         corev1listers.SecretLister
-	configStore          reconciler.ConfigStore
-
-	tracker tracker.Interface
 }
 
 // Check that our Reconciler implements controller.Reconciler
 var _ controller.Reconciler = (*Reconciler)(nil)
 
+// newInitializer creates an Ingress Reconciler and returns ReconcilerInitializer
+func newInitializer(ctx context.Context, cmw configmap.Watcher) ing.ReconcilerInitializer {
+	clusterIngressInformer := clusteringressinformer.Get(ctx)
+	r := &Reconciler{
+		BaseIngressReconciler: ing.NewBaseIngressReconciler(ctx, controllerAgentName, cmw),
+		clusterIngressLister:  clusterIngressInformer.Lister(),
+	}
+	return r
+}
+
+// SetTracker assigns the Tracker field
+func (c *Reconciler) SetTracker(tracker tracker.Interface) {
+	c.Tracker = tracker
+}
+
+// Init method performs initializations to ingress reconciler
+func (c *Reconciler) Init(ctx context.Context, cmw configmap.Watcher, impl *controller.Impl) {
+
+	ing.CommonInitializations(ctx, cmw, c, impl)
+
+	c.Logger.Info("Setting up Ingress event handlers")
+	clusterIngressInformer := clusteringressinformer.Get(ctx)
+
+	myFilterFunc := reconciler.AnnotationFilterFunc(networking.IngressClassAnnotationKey, network.IstioIngressClassName, true)
+	clusterIngressHandler := cache.FilteringResourceEventHandler{
+		FilterFunc: myFilterFunc,
+		Handler:    controller.HandleAll(impl.Enqueue),
+	}
+	clusterIngressInformer.Informer().AddEventHandler(clusterIngressHandler)
+
+	virtualServiceInformer := virtualserviceinformer.Get(ctx)
+	virtualServiceInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: myFilterFunc,
+		Handler:    controller.HandleAll(impl.EnqueueLabelOfClusterScopedResource(networking.ClusterIngressLabelKey)),
```

Comment:
Since Ingress's child resources live in its own namespace, I think we need to do
```
controller.HandleAll(impl.EnqueueControllerOf),
```
instead

That would remove the need for the new keys.

Reply:
This file (clusteringress.go) is related to the ClusterIngress controller.  So, its VSes are still in knative-serving namespace.   The Ingress controller's counterpart can be found in pkg\reconciler\ingress\ingress.go line 128. 
`Handler: controller.HandleAll(impl.EnqueueLabelOfNamespaceScopedResource(networking.IngressNamespaceLabelKey,
			networking.IngressLabelKey)),`
`impl.EnqueueControllerOf` will get the job done and I can eliminate these new label keys, right?

Reply:
I commented on the wrong line.  Yes, we should replace

```controller.HandleAll(impl.EnqueueLabelOfNamespaceScopedResource(networking.IngressNamespaceLabelKey, networking.IngressLabelKey))```

with usage of `impl.EnqueueControllerOf`